src/ansys/fluent/core/session_meshing.py

```diff
@@ -42,59 +42,79 @@ def __init__(
             transcript can be subsequently started and stopped
             using method calls on the ``Session`` object.
         """
+        self.switched = False
         super(Meshing, self).__init__(
             fluent_connection=fluent_connection,
             scheme_eval=scheme_eval,
             file_transfer_service=file_transfer_service,
             start_transcript=start_transcript,
             launcher_args=launcher_args,
         )
-        self.switch_to_solver = lambda: self._switch_to_solver()
-        self.switched = False
 
-    def _switch_to_solver(self) -> Any:
+    def switch_to_solver(self) -> Any:
+        """Switch to solver mode and return a solver session object. Deactivate this
+        object's public interface and streaming services.
+
+        Returns
+        -------
+        Solver
+        """
+        for obj in (
+            self._datamodel_events,
+            self.transcript,
+            self.events,
+            self.monitors,
+        ):
+            obj.stop()
```

Comment:
These callbacks are collected within self._fluent_connection.finalizer_cbs. We could call:
```
for cb in self._fluent_connection.finalizer_cbs:
    cb()
```
Above will also stop the datamodel state streaming and unsubscribe datamodel events.